Modules/timemodule.c

```diff
@@ -69,7 +69,8 @@ check_ticks_per_second(long tps, const char *context)
 {
     /* Effectively, check that _PyTime_MulDiv(t, SEC_TO_NS, ticks_per_second)
        cannot overflow. */
-    if (tps >= 0 && (_PyTime_t)tps > _PyTime_MAX / SEC_TO_NS) {
+    _PyTime_t max_time_in_seconds = _PyTime_MAX / SEC_TO_NS;
+    if (tps >= 0 && (_PyTime_t)tps > max_time_in_seconds) {
```

Comment:
I dislike this fix. You should use LONG_MAX and the preprocessor instead, something like:

```c
#ifdef LONG_MAX >= _PyTime_MAX
if (...)
#endif
```

The compiler is correct that with 32-bit long (which is the case on Windows 64-bit), the condition is always false:

```py
$ python
>>> SEC_TO_NS = 10**9
# _PyTime_t is always signed 64-bit
>>> _PyTime_MAX=2**63-1

# 64-bit signed long
>>> LONG_MAX=2**63-1
>>> LONG_MAX > (_PyTime_MAX // SEC_TO_NS)
True

# 32-bit long
>>> LONG_MAX=2**31-1
>>> LONG_MAX > (_PyTime_MAX // SEC_TO_NS)
False
```